src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1509,16 +1526,18 @@ static int client_parse_message(
         }
 
         r = dhcp_option_parse(message, len, dhcp_lease_parse_options, lease, &error_message);
+        if (client->bootp && r == -ENOMSG)
+                r = DHCP_ACK; // BOOTP messages don't have a DHCP message type option
         if (r < 0)
```

Comment:
prefix with `else` now? otherwise surprising you check a value you already have overwritten again